src/sqlfluff/core/linter.py

```diff
@@ -564,9 +573,19 @@ def persist_changes(self, formatter=None, **kwargs):
 class Linter:
     """The interface class to interact with the linter."""
 
-    def __init__(self, sql_exts=(".sql",), config=None, formatter=None):
-        if config is None:
-            raise ValueError("No config object provided to linter!")
+    def __init__(self, sql_exts=(".sql",), config=None, formatter=None, dialect=None):
+        if dialect and config:
+            raise ValueError(
+                "Cannot specify `dialect` and `config`. Any config object "
+                "specifies it's own dialect."
```

Comment:
Nitpick: No apostrophe is needed in possessive "its": "it's" is short for "it is".

Reply:
🤦